src/Payments.Mvc/Controllers/SearchController.cs

```diff
@@ -20,15 +22,43 @@ public SearchController(ApplicationDbContext dbContext)
 
         public IActionResult Query(string q)
         {
+            if (string.IsNullOrWhiteSpace(q))
+            {
+                ViewBag.ErrorMessage = "No search value entered";
+                return View(new SearchResultsViewModel(){Invoices = new List<Invoice>()});
+            }
+
+            q = q.Trim();
             // get all invoices for team
             var query = _dbContext.Invoices
                 .AsQueryable()
                 .Include(i => i.Items)
                 .Where(i => i.Team.Slug == TeamSlug);
 
+            if (q.Contains("@"))
+            {
+                query = query.Where(a => a.CustomerEmail.Equals(q, StringComparison.OrdinalIgnoreCase));
+            }
+            else
+            {
```

Comment:
Maybe throw in a comment here since it's hard to remember what this path is for (invoice num)